src/main/java/com/google/collegeplanner/servlets/PlannerServlet.java

```diff
@@ -29,35 +35,209 @@
 /** Servlet that returns the mutli-semester plan for the given courses.*/
 @WebServlet("/api/planner")
 public class PlannerServlet extends HttpServlet {
+  private HashMap<String, Integer> indegree; // Number of prerequisites for a course
+  private HashMap<String, HashSet<String>>
+      nextCourses; // Courses that come after the given course in the graph
+  private HashMap<String, HashSet<String>> corequisites;
+  private HashMap<String, Integer> depths; // depth of a course in the graph
+  private HashMap<String, Integer> credits; // Number of credits for each course
+  private ArrayList<String> courseList;
+  private int totalCredits;
+
   /**
-   * Organizes courses into the given number of semesters
+   * Organizes courses from POST request into a given number of semesters
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO(ramyabuva): Implement algorithm to split into the number of semesters
     JSONObject body;
+    int numSemesters;
     try {
       body = getBody(request);
-    } catch (ParseException | NullPointerException e) {
+      numSemesters = Integer.parseInt((String) body.get("semesters"));
+    } catch (ParseException | NullPointerException | NumberFormatException e) {
       respondWithError(
           "Invalid body for POST request.", HttpServletResponse.SC_BAD_REQUEST, response);
       return;
     }
-    String strClasses = (String) body.get("selectedClasses");
-    // Returns the courses in a single semester
-    String[][] courses = {strClasses.split(",")};
+    JSONArray selectedClasses = (JSONArray) body.get("selectedClasses");
```

Comment:
it doesn't look like you validated this input. what happens when you get a garbage input? will the program crash (it should return a 400)? are there potential security issues?